### PR TITLE
feat: add getbroadenedSpectrum function

### PR DESCRIPTION
**Pull request by josoriom**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,8 +43,15 @@
...
     "eslint-config-cheminfo-typescript": "^11.1.0",
     "jest": "^29.0.3",
+    "jest-matcher-deep-close-to": "^3.0.2",
     "prettier": "^2.7.1",
     "rimraf": "^3.0.2",
     "typescript": "^4.8.3"
+  },
+  "dependencies": {
+    "cheminfo-build": "^1.1.11",
+    "ml-array-max": "^1.2.4",
```
Comment (lpatiny) [inline package.json]: Deprecated. We should use the methods of ml-spectra-processing

File: src/getbroadenedSpectrum.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { BorderType, fftConvolution } from 'ml-convolution';
+import { getShape1D } from 'ml-peak-shape-generator';
+
+import { yNorm } from './utilities/yNorm';
```
Comment (lpatiny) [inline src/getbroadenedSpectrum.ts]: Should come from ml-spectra-processing

File: src/getbroadenedSpectrum.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+export function getbroadenedSpectrum(array: number[], options: Options = {}) {
```
Comment (lpatiny) [inline src/getbroadenedSpectrum.ts]: Please add the jsdoc

File: README.md [hunk 4]
```diff
@@ -1,25 +1,47 @@
...
+  <img src="image/readme.png">
+</p>
+<p align="center">
+  A tool for line broadening through integral transforms.
+</p>
 
 ## Installation
 
-`$ npm i ml-ml-integral-transforms`
+`$ npm i integral-transforms`
```
Comment (lpatiny) [inline README.md]: should be ml-integral-transforms

**Review by lpatiny — CHANGES_REQUESTED** [review]
Please correct as well pacage.json. Currently it export ml-ml-... (my fault).

Please check the README that it install ml-integral-transforms

I'm not convinced by the name. Do we have many 'integral-platforms' ? Will this package expose many methods ?

File: package.json (hunk 1 above)
Comment (josoriom) [reply package.json]: Let me check

File: package.json [hunk 5]
```diff
@@ -42,8 +43,15 @@
     "eslint": "^8.24.0",
     "eslint-config-cheminfo-typescript": "^11.1.0",
     "jest": "^29.0.3",
+    "jest-matcher-deep-close-to": "^3.0.2",
     "prettier": "^2.7.1",
     "rimraf": "^3.0.2",
     "typescript": "^4.8.3"
+  },
+  "dependencies": {
+    "cheminfo-build": "^1.1.11",
```
Comment (targos) [inline package.json]: `cheminfo-build` should never be in `dependencies`

File: src/getbroadenedSpectrum.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  const kernel = kernelBasis.getData({
+    length: kernelWidth,
+    height,
+  });
+  const result = fftConvolution(array, kernel, 'CONSTANT' as BorderType);
+  return normalized ? yNorm(result) : result;
+}
+
+interface Options {
+  kind?: 'pseudoVoigt' | 'gaussian' | 'lorentzian';
```
Comment (lpatiny) [inline src/getbroadenedSpectrum.ts]: You should reuse the definition from the shape package

**Comment by lpatiny** [comment]
rename to 'integral-transform'

const {integralTransform} from 'integral-transform'

File: package.json [hunk 7]
```diff
@@ -14,6 +14,7 @@
     "lib-esm"
   ],
   "scripts": {
+    "build": "npm run tsc && cheminfo-build",
```
Comment (lpatiny) [inline package.json]: @josoriom did you try to run it locally ? I think you need to specify the entry point and you may also specify a top level variable.

Something like:

https://github.com/mljs/spectra-processing/blob/master/package.json#L14

**Pull request merged**